AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/PictureRoutes.java

```diff
@@ -0,0 +1,80 @@
+package org.softwire.training.api.routes.v1;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.softwire.training.api.models.ErrorCode;
+import org.softwire.training.api.models.FailedRequestException;
+import org.softwire.training.api.models.PictureApiModel;
+import org.softwire.training.db.daos.PicturesDao;
+import org.softwire.training.db.daos.UsersDao;
+import spark.Request;
+import spark.Response;
+
+import javax.inject.Inject;
+import javax.servlet.MultipartConfigElement;
+import javax.servlet.ServletException;
+import javax.servlet.http.Part;
+import javax.sql.rowset.serial.SerialBlob;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Blob;
+import java.util.Optional;
+
+public class PictureRoutes {
+    private final UsersDao usersDao;
+    private final PicturesDao picturesDao;
+
+    @Inject
+    public PictureRoutes(PicturesDao picturesDao, UsersDao usersDao) {
+        this.picturesDao = picturesDao;
+        this.usersDao = usersDao;
+    }
+
+    public PictureApiModel updatePicture(Request req, Response res, int id) throws FailedRequestException, IOException, ServletException {
+        int userId = req.attribute("user_id");
```

Comment:
I called this update picture but would it be better as createOrUpdate, or would it be better to actually separate the two?
Given that the default picture is served as a static file to save DB effort, it seemed better to think of this only as a single transaction, but I don't know if that's the best way?

Reply:
I think it's fine to leave it as a single operation, I'm indifferent about update vs. createOrUpdate. In a DB you might use the word 'upsert' but not much different.